src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -142,11 +142,11 @@ public SemanticModel(IBicepAnalyzer linterAnalyzer, INamespaceProvider namespace
                         // Resource type parameters are a special case, we need to convert to a dedicated
                         // type so we can compare differently for assignment and code generation.
                         var type = new UnresolvedResourceType(resourceType.TypeReference);
-                        outputs.Add(new OutputMetadata(output.Name, type, description));
+                        outputs.Add(new OutputMetadata(output.Name, type, description, output.DeclaringOutput.HasSecureDecorator));
                     }
                     else
                     {
-                        outputs.Add(new OutputMetadata(output.Name, output.Type, description));
+                        outputs.Add(new OutputMetadata(output.Name, output.Type, description, output.DeclaringOutput.HasSecureDecorator));
```

Comment:
For consistency, should this be using `FindPossibleSecretsVisitor.FindPossibleSecretsInExpression`, rather than checking for the decorator?